### PR TITLE
Leads list service

### PR DESCRIPTION
**Pull request by vmkevv**
[#53911](https://yourday.app/projects/c7a44e83-22e5-47cf-901e-29209beb4161/road-map?taskDetails=2727c097-0206-4783-a6fb-651648508dab) - Tarea

### CODE REVIEW

**Review by marcelo-yourday — CHANGES_REQUESTED** [review]

File: src/models/Lead.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export default class Lead {
+  name: string;
+  email: string;
+  phone: string;
+  country: string;
```
[the remark this answers is not in the document]
Comment (vmkevv) [reply src/models/Lead.ts]: corregido

File: src/models/Lead.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  grade: string;
+  city: string;
+  language: string;
+  createdAt: string;
```
[the remark this answers is not in the document]
Comment (vmkevv) [reply src/models/Lead.ts]: Chaged to Date

File: src/models/Lead.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  constructor() {
+    this.name = '';
+    this.email = '';
+    this.phone = '';
+    this.phone = '';
```
[the remark this answers is not in the document]
Comment (vmkevv) [reply src/models/Lead.ts]: fixed

File: src/handlers/leads.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { RequestHandler } from 'express';
+
+import Lead from '../models/Lead';
+import fileStore from '../persistence/file/fileStorage';
+import { fileNames } from '../persistence/constants';
+
+const leadsHandler: RequestHandler = async (_, res, next) => {
+  try {
+    await fileStore.createIfNotExist(fileNames.leads);
+    const rawText = await fileStore.readContent(fileNames.leads);
```
[the remark this answers is not in the document]
Comment (vmkevv) [reply src/handlers/leads.ts]: Created LeadsRepository class

File: src/models/Lead.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+export default class Lead {
```
[the remark this answers is not in the document]
Comment (vmkevv) [reply src/models/Lead.ts]: filename changed

File: src/index.ts [hunk 6]
```diff
@@ -1,15 +1,12 @@
...
-  const headers = req.headers;
-  console.log(headers);
-  res.statusCode = 200;
-  res.setHeader('Content-Type', 'text/html');
-  res.end('<h1>Hello, World!</h1>');
-});
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.get('/leads', leadsHandler);
```
[the remark this answers is not in the document]
Comment (vmkevv) [reply src/index.ts:8]: Yes, for every new resource we have, we need to add **just** one line of code, I think that is not a big deal. However I'm open to new suggestions.

File: src/persistence/constants.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+const storagePath = `${__dirname}/storage/`;
```
[the remark this answers is not in the document]
Comment (vmkevv) [reply src/persistence/constants.ts:1]: constants.ts deleted

File: src/index.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (ElvisMiranda) [reply src/index.ts:9]: Hi @marcelo-yourday / @eencinasq. This file will certainly grow, what would be your advise in order to fix it and have this PR fixed/merged?

**Review by eencinasq — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (ElvisMiranda) [reply src/index.ts:9]: Oh, that's a great advice @eencinasq, let me work on that, thanks for sharing the link,

[the remark this answers is not in the document]
Comment (vmkevv) [reply src/index.ts]: Leads router created

**Review by eencinasq — APPROVED** [review]

**Pull request merged**